lib/engine/step/special_track.rb

```diff
@@ -35,17 +35,18 @@ def process_lay_tile(action)
       end
 
       def available_hex(entity, hex)
-        return unless ability(entity).hexes.include?(hex.id)
+        return if ability(entity).hexes.any? && !ability(entity).hexes.include?(hex.id)
 
         @game.hex_by_id(hex.id).neighbors.keys
       end
 
       def potential_tiles(entity, hex)
         return [] unless (tile_ability = ability(entity))
 
-        tile_ability
-          .tiles
-          .map { |name| @game.tiles.find { |t| t.name == name } }
+        tiles = tile_ability.tiles.map { |name| @game.tiles.find { |t| t.name == name } }
+        tiles = @game.tiles.uniq(&:name) if tiles.empty?
```

Comment:
this seems like it could be bug prone. what happens if you've run out of tiles, then you allow all?

instead shouldn't you check if tile_abilities doesn't have any tiles?

Reply:
Check tile_ability.tiles instead.